bundles/org.openhab.binding.carnet/src/main/java/org/openhab/binding/carnet/internal/handler/CarNetVehicleHandler.java

```diff
@@ -0,0 +1,643 @@
+                updatedThing.withChannel(channel);
+                created = true;
+            }
+        }
+
+        updateThing(updatedThing.build());
+        return created;
+    }
+
+    private String getReason(CarNetApiErrorDTO error) {
+        CNErrorMessage2Details details = error.details;
+        if (details != null) {
+            return getString(details.reason);
+        }
+        return "";
+    }
+
+    private String getApiStatus(String errorMessage, String errorClass) {
+        if (errorMessage.contains(errorClass)) {
+            // extract the error code like VSR.security.9007
+            String key = API_STATUS_MSG_PREFIX
+                    + substringBefore(substringAfterLast(errorMessage, API_STATUS_CLASS_SECURUTY + "."), ")").trim();
+            return resources.get(key);
+        }
+        return "";
+    }
+
+    private String getError(CarNetException e) {
+        CarNetApiResult res = e.getApiResult();
+        if (res.httpCode == HttpStatus.FORBIDDEN_403) {
+            logger.info("{}: API Service is not available: ", thingId);
+            return "";
+        }
+
+        String reason = "";
+        CarNetApiErrorDTO error = e.getApiResult().getApiError();
+        if (!error.isError()) {
+            return getString(e.getMessage());
+        } else {
+            logger.info("{}: API Call failed: {}", thingId, getString(e.getMessage()));
+            reason = getReason(error);
+            if (!reason.isEmpty()) {
+                logger.debug("{}: {}", thingId, reason);
+            }
+        }
+        if (error.isSecurityClass()) {
+            String message = getApiStatus(error.description, API_STATUS_CLASS_SECURUTY);
+            logger.debug("{}: {}({})", thingId, message, error.description);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, message);
+        }
+
+        CarNetApiResult http = e.getApiResult();
+        if (!http.isHttpOk() && !http.response.isEmpty()) {
+            logger.debug("{}: HTTP response: {}", thingId, http.response);
+        }
+        String message = "";
+        if (!error.code.isEmpty()) {
+            String msgId = API_STATUS_MSG_PREFIX + "." + error.code;
+            message = resources.get(msgId);
+            if (message.equals(msgId)) {
+                // No user friendly message for this code was found, so output the raw description
+                message = message + " - " + error.description;
+            }
+        }
+        logger.debug("{}: {}", thingId, message);
+        return message;
+    }
+
+    private String getChannelAttribute(String channelId, String attribute) {
+        String key = "channel-type.carnet." + channelId + "." + attribute;
+        String value = resources.getText(key);
+        return !value.equals(key) ? value : "";
+    }
+
+    private boolean addService(boolean add, String serviceId, CarNetVehicleBaseService service) {
+        if (add && !services.containsKey(serviceId)) {
```

Comment:
```suggestion
        if (add) {
```

Reply:
nope, I need a true or false as a result if the entry was added